### ISSUE TITLE
Graph construction with Bool features

### ISSUE THREAD

**Issue by Ishitori** [awslabs/graphstorm#521]
I am constructing graph using gconstruct and I download my data which contains boolean features. There features are the result of one-hot-encoding and they truly can be either 0 or 1.

During the call of gconstruct, I get an exception, that boolean type is not supported (`graphstorm/gconstruct/transform.py", line 664`):
```
       assert np.issubdtype(feats.dtype, np.integer) \
                or np.issubdtype(feats.dtype, np.floating), \
                f"The feature {self.feat_name} has to be integers or floats."
```

I can easily fix that problem by adding an extra check for a value to be of boolean type:

```
       assert np.issubdtype(feats.dtype, np.integer) \
                or np.issubdtype(feats.dtype, np.floating) \
                or np.issubdtype(feats.dtype, np.bool_), \
                f"The feature {self.feat_name} has to be integers or floats or bools."

```

The question if this is a valid thing to do? I don't want to recompress data back to int32 because for one-hot-encoding of multiple categories that is wasteful in terms of disk space and networking, but I am not 100% sure that graphstorm does transformation right after adding this condition.

Can someone check if this fix is correct and maybe merge into the main branch?

Thank you.

**Comment by classicsong** [comment]
You can recompress data into int8.
I think bool_ is actually valid in this case.

**Comment by Ishitori** [comment]
Yeah, that what I ended up with - converting data from boolean to Byte type, which is not ideal. But I guess bool has a very special meaning for numpy which is different from regular numeric types.
